flang/test/Lower/array-expression.f90

```diff
@@ -426,4 +426,18 @@ subroutine test18
   row_i => array(i, :)
 end subroutine test18
 
+! CHECK-LABEL: func @_QPtest_column_and_row_order(
+subroutine test_column_and_row_order(x)
+  real :: x(2,3)
+  ! CHECK-DAG: %[[c2:.*]] = fir.convert %c2{{.*}} : (i64) -> index
+  ! CHECK-DAG: %[[column_extent:.*]] = subi %[[c2]], %c1{{.*}} : index
+  ! CHECK-DAG: %[[c3:.*]] = fir.convert %c3{{.*}} : (i64) -> index
+  ! CHECK-DAG: %[[row_extent:.*]] = subi %[[c3]], %c1{{.*}} : index
+  ! CHECK: fir.do_loop %[[row:.*]] = %c0{{.*}} to %[[row_extent]]
+    ! CHECK: fir.do_loop %[[column:.*]] = %c0{{.*}} to %[[column_extent]]
+      ! CHECK: = fir.array_update %{{.*}}, %{{.*}}, %[[column]], %[[row]] : (!fir.array<2x3xf32>, f32, index, index) -> !fir.array<2x3xf32>
```

Comment:
nit: spacing

Reply:
It was on purpose because fir.loop are printed with indentation so I thought it would be nice the reflect this structure. But other lit tests are not doing that, so you are right that it looks weird. I have flattened it.